### PR TITLE
[WNIP] Cache cli config options in config file

### PR DESCRIPTION
**Pull request by sutartmelson**
Addressees #2409 

The config file looks something like this
```
[CLI]
api_url = http://localhost:9080/api/v1
api_key =
username = girder
password = girder
host =
scheme =
port =
api_root =
```

### CODE REVIEW

File: clients/python/girder_client/cli.py [hunk 1]
```diff
@@ -163,6 +177,16 @@ def main(ctx, username, password, api_key, api_url, scheme, host, port, api_root
...
+    if config:
+        ctx.params.update(_parseConfigFile(config))
+        username = ctx.params.get('username')
+        password = ctx.params.get('password')
+        api_key = ctx.params.get('api_key')
+        api_url = ctx.params.get('api_url')
+        scheme = ctx.params.get('scheme')
+        host = ctx.params.get('host')
+        port = ctx.params.get('port')
+        api_root = ctx.params.get('api_root')
```
Comment (zachmullen) [inline clients/python/girder_client/cli.py]: If someone passes `--config` but also passes one of these options explicitly, shouldn't the explicit option override the config file one?

Comment (sutartmelson) [reply clients/python/girder_client/cli.py]: That sounds like useful behavior for the user. I'll do that.

File: clients/python/girder_client/cli.py [hunk 2]
```diff
@@ -113,6 +114,15 @@ def format_options(self, ctx, formatter):
         self.format_commands(ctx, formatter)
 
 
+def _parseConfigFile(path):
```
Comment (danlamanna) [inline clients/python/girder_client/cli.py]: Our existing configuration is ini-style files so we should probably remain consistent with that.

Comment (sutartmelson) [reply clients/python/girder_client/cli.py]: Should this be added to an existing config file somewhere or pass in the path of a cli specific config file?

Comment (danlamanna) [reply clients/python/girder_client/cli.py]: @zachmullen can chime in here, but I think we should probably copy the configuration path precendence set here: https://github.com/girder/girder/blob/61aa5833e91b43d1ea03fa9506175089/girder/utility/config.py#L43-L50 (with a differently named configuration file).

Comment (zachmullen) [reply clients/python/girder_client/cli.py]: 👍  

1. Value of `--config` option if passed.
2. Value of `GIRDER_CLIENT_CONFIG` env var if set.
3. ~/.girder/girder_client.cfg
4. /etc/girder_client.cfg

File: clients/python/girder_client/cli.py [hunk 3]
```diff
@@ -177,21 +197,29 @@ def main(ctx, username, password, api_key, api_url, scheme, host, port, api_root
     input his/her password.
     """
     # --api-url and URL by part arguments are mutually exclusive
+    if not config:
+        paths = _loadConfigPaths()
+        if paths:
+            config = paths[0]
```
Comment (zachmullen) [inline clients/python/girder_client/cli.py]: I'm not sure whether the behavior here should be to only look at one config file, or just list them all in order of precedence and have ConfigParser attempt to read all of them. I'm leaning toward the latter, which would eliminate a lot of this custom logic.

**Comment by jcfr** [comment]
I suggest to also add tests.

**Pull request closed without merging**